ts/identifier.go

```diff
@@ -25,14 +25,16 @@ import (
 	"crypto/md5"
 )
 
-// BinaryID constructs a new ID based on a binary value
+// BinaryID constructs a new ID based on a binary value.
+// WARNING: Does not copy the underlying data, do not use
+// when cloning a pooled ID object.
 func BinaryID(v []byte) ID {
-	return &id{data: append(make([]byte, 0, len(v)), v...)}
+	return &id{data: v}
```

Comment:
Might it be better to keep the copy semantics but just always clone instead of `id.BinaryID(otherID.Data())`?  That way we enforce semantics and you can use `Clone()` to hit the fast path.